perf/src/sigverify.rs

```diff
@@ -496,6 +503,46 @@ impl Deduper {
     }
 }
 
+//inplace shrink a batch of packets
+pub fn shrink_batches(batches: &mut Vec<PacketBatch>) -> usize {
+    let mut keep_batch_index = 0;
+    let mut keep_packet_index = 0;
+    let mut shrunken_length = 0;
+    for batch_index in 0..batches.len() {
+        for packet_index in 0..batches[batch_index].packets.len() {
+            if batches[batch_index].packets[packet_index].meta.discard() {
+                continue;
+            }
+            'concentrate_kept: while keep_batch_index < batch_index {
```

Comment:
tbh, this is way more confusing, label + loop is really hard to grok

Reply:
I understand that the pattern is somewhat novel, but if you look at it in context of the [commit](https://github.com/aeyakovenko/solana/pull/333/commits/cb37843aaa32e30a2d3afd5543e65d668013ca89) that made the change, you'll see that it reduces the complexity by at least three paths within nested loops.  Code like this is highly susceptible to introducing bugs upon modification (if it didn't already have some).

Reply:
the while/while continue/break is standard way to do deal with multiple iterators that are not stepping at the same time C. this is just confusing af.